### ISSUE TITLE
Entry-Key Format

### ISSUE THREAD

**Issue by kellertuer** [thchr/DOI2Bibtex.jl#5]
Sorry that I directly have 2 feature requests.
Since there is already the nice Abbreviation method, would it be possible to do something similar with a Key-generator?
My favourite key name is `AllLastNamesOfAuthorsCamelCased:YYYY`.

I am not 100& sure how one could set this (maybe different styles or even keeping the default?) but it would be a neat idea if there is a way to do so.

**Comment by thchr** [comment]
This is a bit tricky because I currently just resort to a few regexes for extracting authors, year, and title.

In principle, I ought to be doing this with a BibTeX parser but I didn't want to take on more dependencies than necessary - i.e., I just wanted to return a string that could then potentially be parsed by some Julia implementation of a BibTeX parser later (I use my own unregistered one - https://github.com/thchr/SimpleBibTeX.jl - but I know there many other, probably much more mature ones).

Without a proper BibTeX parser, extending this is probably going to be fragile and quite constrained. I.e., for now, I'll probably punt on this.

**Comment by kellertuer** [comment]
Right, it is a bit tricky and for best of cases we have a Bib(La)TeX parser for that.

Your Parser looks nice and simple and seems to also allow for BibLaTeX directly. If you do not mind I might try to find time to document it a bit (I grew kind of into a documentation nerd).
